rust_src/src/vector_macros.rs

```diff
@@ -33,3 +42,19 @@ macro_rules! allocate_pseudovector {
         }
     };
 }
+
+// Equivalent to `ALLOCATE_ZEROED_PSEUDOVECTOR` in C
+/// Allocate fully initialized pseudovector where all Lisp_Object
+/// slots are set to Qnil and the rest (if any) is zeroed.
```

Comment:
It's often quite ambiguous what macros expect as arguments, it would be nice if you could add a description of what arguments the macro expects and/or an example invocation in the doc comment.